error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/AssertJPathRules.java

```diff
@@ -0,0 +1,216 @@
+    }
+  }
+
+  static final class AbstractPathAssertIsRegularFile {
+    @BeforeTemplate
+    AbstractBooleanAssert<?> before(Path path) {
+      return assertThat(Files.isRegularFile(path)).isTrue();
+    }
+
+    @AfterTemplate
+    AbstractPathAssert<?> after(Path path) {
+      return assertThat(path).isRegularFile();
+    }
+  }
+
+  static final class AbstractPathAssertIsDirectory {
+    @BeforeTemplate
+    AbstractBooleanAssert<?> before(Path path) {
+      return assertThat(Files.isDirectory(path)).isTrue();
+    }
+
+    @AfterTemplate
+    AbstractPathAssert<?> after(Path path) {
+      return assertThat(path).isDirectory();
+    }
+  }
+
+  static final class AbstractPathAssertIsSymbolicLink {
+    @BeforeTemplate
+    AbstractBooleanAssert<?> before(Path path) {
+      return assertThat(Files.isSymbolicLink(path)).isTrue();
+    }
+
+    @AfterTemplate
+    AbstractPathAssert<?> after(Path path) {
+      return assertThat(path).isSymbolicLink();
+    }
+  }
+
+  static final class AbstractPathAssertIsAbsolute {
+    @BeforeTemplate
+    AbstractBooleanAssert<?> before(Path path) {
+      return assertThat(path.isAbsolute()).isTrue();
+    }
+
+    @AfterTemplate
+    AbstractPathAssert<?> after(Path path) {
+      return assertThat(path).isAbsolute();
+    }
+  }
+
+  static final class AbstractPathAssertIsRelative {
+    @BeforeTemplate
+    AbstractBooleanAssert<?> before(Path path) {
+      return assertThat(path.isAbsolute()).isFalse();
+    }
+
+    @AfterTemplate
+    AbstractPathAssert<?> after(Path path) {
+      return assertThat(path).isRelative();
+    }
+  }
+
+  static final class AbstractPathAssertIsReadable {
+    @BeforeTemplate
+    AbstractBooleanAssert<?> before(Path path) {
+      return assertThat(Files.isReadable(path)).isTrue();
+    }
+
+    @AfterTemplate
+    AbstractPathAssert<?> after(Path path) {
+      return assertThat(path).isReadable();
+    }
+  }
+
+  static final class AbstractPathAssertIsWritable {
+    @BeforeTemplate
+    AbstractBooleanAssert<?> before(Path path) {
+      return assertThat(Files.isWritable(path)).isTrue();
+    }
+
+    @AfterTemplate
+    AbstractPathAssert<?> after(Path path) {
+      return assertThat(path).isWritable();
+    }
+  }
+
+  static final class AbstractPathAssertIsExecutable {
+    @BeforeTemplate
+    AbstractBooleanAssert<?> before(Path path) {
+      return assertThat(Files.isExecutable(path)).isTrue();
+    }
+
+    @AfterTemplate
+    AbstractPathAssert<?> after(Path path) {
+      return assertThat(path).isExecutable();
+    }
+  }
+
+  static final class AbstractPathAssertHasFileName {
+    @BeforeTemplate
+    @SuppressWarnings("AssertThatHasToString")
+    AbstractStringAssert<?> before(Path path, String fileName) {
+      return assertThat(path.getFileName().toString()).isEqualTo(fileName);
+    }
+
+    @AfterTemplate
+    AbstractPathAssert<?> after(Path path, String fileName) {
+      return assertThat(path).hasFileName(fileName);
+    }
+  }
```

Comment:
Calling out the `@SuppressWarnings` here for review scrutiny: We're converting to `hasFileName` here instead of `hasToString`.

Alternatively we could (instead or also) convert
`assertThat(path.getFileName()).hasToString(fileName)` to 
`assertThat(path).hasFileName(fileName)`.